hlapipeline/alignimages.py

```diff
@@ -3,65 +3,173 @@
 
     Returns
     -------
     Nothing for now.
 
     """
 
+    # Define astrometric catalog list in priority order
+    catalogList = ['GAIADR2', 'GSC241']
+    numCatalogs = len(catalogList)
+
     # 1: Interpret input data and optional parameters
+    print("-------------------- STEP 1: Get data --------------------")
+    imglist = check_and_get_data(input_list)
+    print("\nSUCCESS")
 
     # 2: Apply filter to input observations to insure that they meet minimum criteria for being able to be aligned
-    imglist = input_list
+    print("-------------------- STEP 2: Filter data --------------------")
+    filteredTable = filter.analyze_data(imglist)
+
+    # Check the table to determine if there is any viable data to be aligned.  The
+    # 'doProcess' column (bool) indicates the image/file should or should not be used
+    # for alignment purposes.
+    if filteredTable['doProcess'].sum() == 0:
+        print("No viable images in filtered table - no processing done.\n")
+        return
+
+    # Get the list of all "good" files to use for the alignment
+    processList = filteredTable['imageName'][np.where(filteredTable['doProcess'])]
+    processList = list(processList) #Convert processList from numpy list to regular python list
+    print("\nSUCCESS")
 
     # 3: Build WCS for full set of input observations
-    refwcs = amutils.build_reference_wcs(imglist)
+    print("-------------------- STEP 3: Build WCS --------------------")
+    refwcs = amutils.build_reference_wcs(processList)
+    print("\nSUCCESS")
 
     # 4: Retrieve list of astrometric sources from database
-    reference_catalog = generate_astrometric_catalog(imglist, catalog='GAIADR2')
-
+    print("-------------------- STEP 4: Detect astrometric sources --------------------")
+    # While loop to accommodate using multiple catalogs
+    doneFitting = False
+    catalogIndex = 0
+    while not doneFitting:
+        reference_catalog = generate_astrometric_catalog(processList, catalog='GAIADR2', existing_wcs=refwcs)
+
+        # The table must have at least MIN_CATALOG_THRESHOLD entries to be useful
+        if len(reference_catalog) < MIN_CATALOG_THRESHOLD:
+            print("Not enough sources found in catalog" + catalogList[catalogIndex])
+            catalogIndex += 1
+            if catalogIndex <= (numCatalogs - 1):
+                print("Try again with the next catalog")
+                continue
+            else:
+                print("Not enough sources found in any catalog - no processing done.")
+                return
+        else:
+            doneFitting = True
+    print("\nSUCCESS")
     # 5: Extract catalog of observable sources from each input image
-    extracted_sources = generate_source_catalogs(imglist, refwcs, threshold=1000)
-
+    print("-------------------- STEP 5: Source finding --------------------")
+    extracted_sources = generate_source_catalogs(processList, refwcs, threshold=1000)
```

Comment:
The specification of the ``threshold`` parameter here should be removed so that the function ``generate_source_catalogs`` can rely on the detector-specific parameters defined for each detector.  It also turns out (so far) that ``refwcs`` will probably NOT be needed at this point either in order to provide the correct inputs for ``tweakwcs``.  This is something that will probably need to wait until #37 gets merged.

Reply:
I am sure leaving `theshold=1000` here is an oversight as we discussed removing this parameter for the reason Warren sighted.

Reply:
Issue resolved in commit [4ac4d36](https://github.com/spacetelescope/hlapipeline/tree/4ac4d36c00df612e9f5ae8dc82b7e6730f9f75e2)